make/compiler_flags

```diff
@@ -7,6 +7,19 @@ ifneq ($(OS),Windows_NT)
   OS := $(shell uname -s)
 endif
 
+## Set OS specific library filename extensions
+ifeq ($(OS),Windows_NT)
+  LIBRARY_SUFFIX := .dll
+endif
+
+ifeq ($(OS),Darwin)
+  LIBRARY_SUFFIX := .dylib
+endif
+
+ifeq ($(OS),Linux)
+  LIBRARY_SUFFIX := .so
+endif
```

Comment:
I would feel more comfortable with something in the lines of:
```
ifeq ($(OS),Windows_NT)
  LIBRARY_SUFFIX := .dll
else ifeq ($(OS),Darwin)
  LIBRARY_SUFFIX := .dylib
else ifeq ($(OS),Linux)
  LIBRARY_SUFFIX := .so
else
  $(error Can't detect OS properly. This will impede building the TBB dependency of Stan Math. Please file an issue.)
endif
```
I know this case will probably/hopefully never happen but in case that it does, it makes support easier.

Reply:
I think that our makefiles anyway fall over if we are not in any of these 3 cases. So why should we suspect for this check that this is not the case? 

We could add a general check for the os and fail if it’s not one of the expected...but that would rule out to use this on other platforms from which we just don’t know yet that we support them.

Maybe using ?= is better here, since the user can override that?

The tbb supports android, for example...

Reply:
> I think that our makefiles anyway fall over if we are not in any of these 3 cases. So why should we suspect for this check that this is not the case?
> 
> We could add a general check for the os and fail if it’s not one of the expected...but that would rule out to use this on other platforms from which we just don’t know yet that we support them.

Come to think of it, we should not do it here. Otherwise it will interfere with [this check](https://github.com/stan-dev/cmdstan/pull/737/files) that will be added for Stanc3. This one would trigger before that and that might be confusing.

> Maybe using ?= is better here, since the user can override that?

Yes, lets do that.